src/components/ClassSearch.vue

```diff
@@ -52,6 +52,193 @@ import FilterSet from './FilterSet.vue';
+      'atStart': atStart,
+      'asLiteral': asLiteral
+    };
+
+    // Order to apply functions (must add ^ after escaping, for example)
+    var applicationOrder = ['asLiteral', 'atStart'];
+
+    // Get regex from inputs
+    var regexAddOn = '';
+
+    if (this.requires !== undefined) {
+      regexAddOn = inputs[this.requires];
+    }
+
+    var regex = this.regex + regexAddOn;
+
+    // Create list of different combinations of functions to apply
+    // Prioritized functions last; which priority to consider first by priority order
+    var regexPriorities = [[]];
+
+    priorityOrder.reverse();
+    for (var p = 0; p < priorityOrder.length; p++) {
+      var isPrioritized = priorityDirections[priorityOrder[p]];
+      // Create two lists; one with priority applied, the other without
+      var arrayWithPriority = regexPriorities.map((funcNames) => funcNames.concat([priorityOrder[p]]));
+
+      // If this priority is prioritized, put it last
+      if (isPrioritized) {
+        regexPriorities.push(...arrayWithPriority);
+      } else {
+        arrayWithPriority.push(...regexPriorities);
+        regexPriorities = arrayWithPriority;
+      }
+    }
+
+    // Apply these functions to the base regex, in the application order
+    var priorities = regexPriorities.map(function (priorityFuncs) {
+      return priorityFuncs
+        .sort((a, b) => applicationOrder.indexOf(a) - applicationOrder.indexOf(b))
+        .reduce((acc, func) => (priorityFunctions[func])(acc), regex);
+    }).map(RegexFilter.getRegexTestFunction);
+
+    // Set member variable to use in future prioritization checks
+    this.priorities = priorities;
+  }
+
+  compareByVariants (subject) {
+    // List of sort numbers for different attributes
+    var orders = [];
+
+    for (var a = 0; a < this.attributes.length; a++) {
+      // Get last index where the regex in the priority list matches
+      // The priority list is in reverse order of priority
+      var matches = this.priorities.map((test) => test(subject[this.attributes[a]]));
+      orders.push(matches.lastIndexOf(true));
+    }
+
+    // Get largest sort priority of all the matching attributes
+    return Math.max(...orders);
+  }
+}
+
+class MathFilter extends Filter {
+  constructor (name, shortName, range, inclusive, attributeNames, mode) {
+    var comparator = function (input) {
+      if ((range[0] === undefined || input > range[0]) && (range[1] === undefined || input < range[1])) {
+        return true;
+      } else if (inclusive && (input === range[0] || input === range[1])) {
+        return true;
+      }
+      return false;
+    };
+    super(name, shortName, comparator, attributeNames, [], mode);
+  }
+}
+
+class FilterGroup {
+  constructor (name, filters, combination) {
+    this.name = name;
+    this.filters = filters;
+    this.combine = {
+      'AND': (a, b) => a && b,
+      'OR': (a, b) => a || b
+    }[combination];
+  }
+
+  matches (subject, applied, inputs) {
```

Comment:
`applied` could have a better name, it's not clear that this is which filters are active (right?)

Reply:
much better, thanks!